src/cargo/sources/registry/mod.rs

```diff
@@ -213,17 +212,121 @@ pub struct RegistryConfig {
+            Features,
+            Cksum,
+            Yanked,
+            Links,
+        }
+
+        impl<'de> de::Visitor<'de> for Visitor {
+            type Value = RegistryPackage<'de>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                write!(formatter, "struct RegistryPackage")
+            }
+
+            fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::MapAccess<'de>,
+            {
+                let mut name = None;
+                let mut vers = None;
+                let mut deps = None;
+                let mut features = None;
+                let mut cksum = None;
+                let mut yanked = None;
+                let mut links = None;
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::Name => {
+                            if name.is_some() {
+                                return Err(de::Error::duplicate_field("name"));
+                            }
+                            name = Some(map.next_value()?);
+                        }
+                        Field::Vers => {
+                            if vers.is_some() {
+                                return Err(de::Error::duplicate_field("vers"));
+                            }
+                            vers = Some(map.next_value()?);
+                        }
+                        Field::Deps => {
+                            if deps.is_some() {
+                                return Err(de::Error::duplicate_field("deps"));
+                            }
+                            deps =
+                                Some(map.next_value_seed(DependencyListDefault(self.0.clone()))?);
+                        }
+                        Field::Features => {
+                            if features.is_some() {
+                                return Err(de::Error::duplicate_field("features"));
+                            }
+                            features = Some(map.next_value()?);
+                        }
+                        Field::Cksum => {
+                            if cksum.is_some() {
+                                return Err(de::Error::duplicate_field("cksum"));
+                            }
+                            cksum = Some(map.next_value()?);
+                        }
+                        Field::Yanked => {
+                            if yanked.is_some() {
+                                return Err(de::Error::duplicate_field("yanked"));
+                            }
+                            yanked = Some(map.next_value()?);
+                        }
+                        Field::Links => {
+                            if links.is_some() {
+                                return Err(de::Error::duplicate_field("links"));
+                            }
+                            links = Some(map.next_value()?);
```

Comment:
I would write `links = map.next_value()?` and then use `RegistryPackage { ..., links }` below without an `unwrap_or(None)`, but this is fine if you prefer the symmetry with the other fields.